app/services/switch_resource_type.rb

```diff
@@ -0,0 +1,49 @@
+class SwitchResourceType
+  class << self
+    def switch_resource(from_resource, to_resource, from_resource_id)
+      if from= from_resource.find(from_resource_id).destroy
```

Comment:
Can we have the assignment _cum_ argument to *if* evaluated on a separate line then you might want to consider when `resource` is not found :thought_balloon: